contents/ghcup.mdx

```diff
@@ -7,6 +7,8 @@ GHCup 是一种用于安装 Haskell 的工具，它使得用户可以轻易地
 
 GHCup 类似 Rustup，可以用于安装 Haskell 工具链。建议搭配 Hackage 和 Stackage 源使用。
 
+> 如果想要使用 XDG 规范，可以设置 `GHCUP_USE_XDG_DIRS` 变量，例如 `export GHCUP_USE_XDG_DIRS=1`
```

Comment:
1. Windows 与 macOS 下这个环境变量是否有效？
2. 如果设置了这个环境变量，下面配置的路径会变成什么？

Reply:
> 1. Windows 与 macOS 下这个环境变量是否有效？
> 2. 如果设置了这个环境变量，下面配置的路径会变成什么？

https://www.haskell.org/ghcup/guide/#xdg-support

关于 Windows 以及具体路径变化，请自行查询 XDG 规范

Reply:
经过我核实，这个选项确实只对非 Windows（即 MINGW 或 MSYS2）有效

```shell
# ...
case "${plat}" in
        MSYS*|MINGW*|CYGWIN*)
			: "${GHCUP_INSTALL_BASE_PREFIX:=/c}"
			GHCUP_DIR=$(cygpath -u "${GHCUP_INSTALL_BASE_PREFIX}/ghcup")
			GHCUP_BIN=$(cygpath -u "${GHCUP_INSTALL_BASE_PREFIX}/ghcup/bin")
			: "${GHCUP_MSYS2:=${GHCUP_DIR}/msys64}"
			set_msys2_env_dir
			;;
		*)
			: "${GHCUP_INSTALL_BASE_PREFIX:=$HOME}"

			if [ -n "${GHCUP_USE_XDG_DIRS}" ] ; then
				GHCUP_DIR=${XDG_DATA_HOME:=$HOME/.local/share}/ghcup
				GHCUP_BIN=${XDG_BIN_HOME:=$HOME/.local/bin}
			else
				GHCUP_DIR=${GHCUP_INSTALL_BASE_PREFIX}/.ghcup
				GHCUP_BIN=${GHCUP_INSTALL_BASE_PREFIX}/.ghcup/bin
			fi
			;;
esac
# ...
```

对应代码也是

```haskell
-- | ~/.ghcup by default
--
-- If 'GHCUP_USE_XDG_DIRS' is set (to anything),
-- then uses 'XDG_DATA_HOME/ghcup' as per xdg spec.
ghcupBaseDir :: IO GHCupPath
ghcupBaseDir
  | isWindows = do
      bdir <- fromMaybe "C:\\" <$> lookupEnv "GHCUP_INSTALL_BASE_PREFIX"
      pure (GHCupPath (bdir </> "ghcup"))
  | otherwise = do
      xdg <- useXDG
      if xdg
        then do
          bdir <- lookupEnv "XDG_DATA_HOME" >>= \case
            Just r  -> pure r
            Nothing -> do
              home <- liftIO getHomeDirectory
              pure (home </> ".local" </> "share")
          pure (GHCupPath (bdir </> "ghcup"))
        else do
          bdir <- lookupEnv "GHCUP_INSTALL_BASE_PREFIX" >>= \case
            Just r  -> pure r
            Nothing -> liftIO getHomeDirectory
          pure (GHCupPath (bdir </> ".ghcup"))
```

这里我确实要修改一下，说明情况

Reply:
我的建议是：

1. <https://www.haskell.org/ghcup/guide/#xdg-support> 这个链接需要明确包含在帮助中。
2. 帮助的下文中提到了类似 `~/.ghcup/config.yaml` 的路径，一种做法是在提到 XDG 的地方补充下文中出现的路径会被替换成什么。

Reply:
btw，把 xdg 这一段放在 `>` (Quote) 块里面感觉怪怪的，为什么不直接提出来呢？

Reply:
因为我认为 xdg 并不是所有人都需要的，这只是一种补充配置，所以后续配置变动的地方我都会使用 comment 来表述

Reply:
或许也可以单独成一个帮助小章节，我改一下看看